Options.py

```diff
@@ -900,6 +901,217 @@ class ItemSet(OptionSet):
+        return entrance.lower() in cls.entrances
+
+    @classmethod
+    def validate_exit_name(cls, exit: str) -> bool:
+        return exit.lower() in cls.exits
+
+    @classmethod
+    def can_connect(cls, entrance: str, exit: str) -> bool:
+        """Checks that a given entrance can connect to a given exit.
+        By default, this will always return true unless overridden."""
+        return True
+
+    @classmethod
+    def validate_plando_connections(cls, connections: typing.Iterable[PlandoConnection]) -> None:
+        used_entrances: typing.List[str] = []
+        used_exits: typing.List[str] = []
+        for connection in connections:
+            entrance = connection.entrance
+            exit = connection.exit
+            direction = connection.direction
+            if direction not in (PlandoConnections.Direction.Entrance,
+                                 PlandoConnections.Direction.Exit,
+                                 PlandoConnections.Direction.Both):
+                raise ValueError(f"Unknown direction: {direction}")
+            if entrance in used_entrances:
+                raise ValueError(f"Duplicate Entrance {entrance} not allowed.")
+            if exit in used_exits:
+                raise ValueError(f"Duplicate Exit {exit} not allowed.")
+            used_entrances.append(entrance)
+            used_exits.append(exit)
+            if not cls.validate_entrance_name(entrance):
+                raise ValueError(f"{entrance.title()} is not a valid entrance.")
+            if not cls.validate_exit_name(exit):
+                raise ValueError(f"{exit.title()} is not a valid exit.")
+            if not cls.can_connect(entrance, exit):
+                raise ValueError(f"Connection between {entrance.title()} and {exit.title()} is invalid.")
+
+    @classmethod
+    def from_any(cls, data: PlandoConFromAnyType) -> Self:
+        if isinstance(data, typing.Iterable):
+            value: typing.List[PlandoConnection] = []
+            for connection in data:
+                if isinstance(connection, typing.Mapping):
+                    percentage = connection.get("percentage", 100)
+                    if random.random() < float(percentage / 100):
+                        entrance = connection.get("entrance", None)
+                        exit = connection.get("exit", None)
+                        direction = connection.get("direction", "both")
+
+                        if not entrance or not exit:
+                            raise Exception("Plando connection must have an entrance and an exit.")
+                        value.append(PlandoConnection(
+                            entrance,
+                            exit,
+                            direction,
+                            percentage
+                        ))
+                elif isinstance(connection, PlandoConnection):
+                    if random.random() < float(connection.percentage / 100):
```

Comment:
should this be connection.get("percentage", 100)

Reply:
`NamedTuple` doesn't have `get`

Reply:
ah i misread the `isinstance`

Reply:
Besides that point, PlandoConnection will always have percentage, while percentage is optional if converting from dict.

Reply:
i understand i just missed it was a different type there when reading through 😓